lib/proxy_api/external_ipam.rb

```diff
@@ -0,0 +1,127 @@
+module ProxyAPI
+  class ExternalIpam < ProxyAPI::Resource
+    def initialize(args)
+      @url = args[:url] + "/phpipam"
+      super args
+    end
+
+    # Retrieves the next available IP address for the specified subnet
+    # Returns: Hash with next_ip on success, hash with error otherwise
+    #   Example(success): {"cidr":"100.10.10.0/24","next_ip":"100.10.10.23"}
+    #   Example(error):   {"error":"The specified subnet does not exist in phpIPAM."}
+    def next_ip(subnet)
+      cidr = subnet.network + '/' + subnet.cidr.to_s if subnet.network.present?
+      get("/next_ip?cidr=#{cidr}")
+    rescue => e
+      raise ProxyException.new(url, e, N_("Unable to retrieve the next available IP for subnet %s from phpIPAM."), cidr)
+    end
```

Comment:
When users create many hosts concurrently, this can easily run into race condition - a single IP is returned for multiple hosts. I suggest to put MAC as a required parameter and the implementation should memorize what IP it offered to which MAC addresses and never offer the same IP for different MACs. We actually do this in the subnet API in smart proxy - offered IPs are memorized for few minutes. I suggest you take the same approach.

Also document how subnet input param should look like - is that CIDR address?